pkg/quota/evaluator/core/pods.go

```diff
@@ -265,12 +301,12 @@ func toInternalPodOrError(obj runtime.Object) (*api.Pod, error) {
 }
 
 // podMatchesScopeFunc is a function that knows how to evaluate if a pod matches a scope
-func podMatchesScopeFunc(scope api.ResourceQuotaScope, object runtime.Object) (bool, error) {
+func podMatchesScopeFunc(selector api.ScopedResourceSelectorRequirement, object runtime.Object) (bool, error) {
 	pod, err := toInternalPodOrError(object)
 	if err != nil {
 		return false, err
 	}
-	switch scope {
+	switch selector.ScopeName {
```

Comment:
if the new syntax matches an exists operation on the old syntax, the other cases probably need to be updated at a future time.  if you use the new syntax, we lack some protection with what invalidScopePairs did in validation to find conflicting scopes.  if validation ensured that exists operator was only valid for the other types, i think this would keep the code logically consistent.  WDYT?

Reply:
where other types == other scopes (duration and qos related)

Reply:
I think you meant "if the validation ensured that for other types only exists operator was valid, ..." ?

Reply:
updated `validateScopeSelector` in validation.go to ensure that if "other scopes" are being expressed in quota object using scope selector, any existing validation logic is not missed.

Reply:
correct, this looks good now.